.cargo-husky/hooks/pre-commit

```diff
@@ -14,7 +14,7 @@ fi
 set -xe
 
 # Setting RUSTFLAGS env for clippy makes it not include custom rules
-RUSTFLAGS=-Dwarnings cargo check --workspace --all-targets --profile bench ${ALL_FEATURES}
+RUSTFLAGS=-Dwarnings cargo hack check --feature-powerset --no-dev-deps
```

Comment:
This doesn't need to be run in the pre-commit hook. The idea is to make CI more expansive and correct, but keep local checks minimal

Reply:
Hmm are you sure? I find it really beneficial to run the same checks that the CI does (at least for Rust) before opening a PR. I find that it saves me time in the long run due to fewer adjustments and context switching later on. It can also save time for reviewers.

Performance-wise this change leads to 15% slower pre commit hook execution time for a clean build on my machine.

Reply:
As discussed with @Wodann we'll support running full CI checks in the release CLI (https://github.com/NomicFoundation/rethnet/issues/63)

Reverted to `cargo check` in pre-commit in [0cb0dc8](https://github.com/NomicFoundation/hardhat/pull/4380/commits/0cb0dc81d17d19c1f20b00fc0918d2451b727e20)